website_event_sale_check_order_before_payment/controllers/main.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from odoo import _, http
+from odoo.http import request
+
+import odoo.addons.website_sale.controllers.main as website_sale
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteSale(website_sale.WebsiteSale):
+    @http.route(
+        ["/shop/payment/order_validity"], type="json", auth="public", website=True
+    )
+    def check_order_validity(self):
+        order = request.website.sale_get_order()
+        if order:
+            if order.state != "draft":
+                return {
+                    "order_is_valid": False,
+                    "invalid_order_message": request.env.company.cancelled_order_message,
+                }
```

Comment:
```suggestion
                return {
                    "valid": False,
                    "message": request.env.company.cancelled_order_message,
                }
```

Nitpicking: I'd reduce the verbosity of the dict key names, like this.